dealership/contracts.py

```diff
@@ -1,12 +1,75 @@
+from dealership.customers import Customer, Employee
+from dealership.vehicles import Car, Truck, Motorcycle
+
 class Contract(object):
-    pass
+    '''Represents a contract with the dealership. Must be implemented as a
+    BuyContract or a LeaseContract.'''
+    def __init__(self, vehicle, customer, num_payments):
+        self.vehicle = vehicle
+        self.customer = customer
+        self.discount_rate = 0.10
+        
+        if isinstance(self, BuyContract):
+            self.monthly_payments = num_payments
+        elif isinstance(self, LeaseContract):
+            self.length_in_months = num_payments
+        else:
+            raise NotImplementedError()
+    
+    def total_value(self):
+        raise NotImplementedError()
+    
+    def monthly_value(self):
+        raise NotImplementedError()
 
 
 class BuyContract(Contract):
+    '''Represents a contract for purchasing a vehicle.'''
     def __init__(self, vehicle, customer, monthly_payments):
-        pass
+        super(BuyContract, self).__init__(vehicle, customer, monthly_payments)
+        
+        if isinstance(self.vehicle, Car):
+            self.interest_rate = 0.07
+        elif isinstance(self.vehicle, Motorcycle):
+            self.interest_rate = 0.03
+        elif isinstance(self.vehicle, Truck):
+            self.interest_rate = 0.11
+    
```

Comment:
I would just store all of the interests rates as 1.xx as you end up adding 1 to them anyway, and similarly for the discount rate.